packages/react-core/src/components/ProgressStepper/__tests__/ProgressStep.test.tsx

```diff
@@ -0,0 +1,168 @@
+  render(<ProgressStep>Test</ProgressStep>);
+  expect(screen.getByRole('listitem')).not.toHaveAttribute('aria-current');
+});
+
+test(`Renders with a aria-current of "step" on containing li element when isCurrent is passed`, () => {
+  render(<ProgressStep isCurrent>Test</ProgressStep>);
+  expect(screen.getByRole('listitem')).toHaveAttribute('aria-current', 'step');
+});
+
+test(`Renders with the aria label on containing li element passed via prop`, () => {
+  render(<ProgressStep aria-label="Test label">Test</ProgressStep>);
+  expect(screen.getByRole('listitem')).toHaveAccessibleName('Test label');
+});
+
+test('Renders the element passed via the description prop', () => {
+  render(<ProgressStep description={<button>Description</button>}>Test</ProgressStep>);
+  expect(screen.getByRole('button', { name: 'Description' })).toBeVisible();
+});
+
+test('Renders the description element inside div element with class pf-c-progress-stepper__step-description', () => {
+  render(<ProgressStep description="Description">Test</ProgressStep>);
+  expect(screen.getByText('Description')).toHaveClass('pf-c-progress-stepper__step-description');
+});
+
+test('Does not render with aria-labelledby attribute by default', () => {
+  render(<ProgressStep>Test</ProgressStep>);
+  expect(screen.getByText('Test')).not.toHaveAttribute('aria-labelledby');
+});
+
+test('Renders children inside class pf-c-progress-stepper__step-title', () => {
+  render(<ProgressStep>Test</ProgressStep>);
+  expect(screen.getByText('Test')).toHaveClass('pf-c-progress-stepper__step-title');
+});
+
+test('Renders Component element as div by default', () => {
+  render(<ProgressStep>Test</ProgressStep>);
+  expect(screen.getByText('Test')).toHaveProperty('nodeName', 'DIV');
+});
+
+test('Renders Component element as button when popoverRender is defined', () => {
+  render(<ProgressStep popoverRender={() => <div>Popover</div>}>Test</ProgressStep>);
+  expect(screen.getByRole('button')).toBeVisible();
+});
+
+test('Does not render with class pf-m-help-text by default on Component element', () => {
+  render(<ProgressStep>Test</ProgressStep>);
+  expect(screen.getByText('Test')).not.toHaveClass('pf-m-help-text');
+});
+
+test('Renders with class pf-m-help-text when popoverRender is defined', () => {
+  render(<ProgressStep popoverRender={() => <div>Popover</div>}>Test</ProgressStep>);
+  expect(screen.getByText('Test')).toHaveClass('pf-m-help-text');
+});
+
+test('Does not renders with aria-labelledBy by default on Component element', () => {
+  render(<ProgressStep>Test</ProgressStep>);
+  expect(screen.getByText('Test')).not.toHaveAttribute('aria-labelledby');
+});
+
+test('Renders with aria-labelledBy on Component element when id and titleId are defined', () => {
+  render(
+    <ProgressStep id="test-id" titleId="title-id">
+      Test
+    </ProgressStep>
+  );
+  expect(screen.getByText('Test')).toHaveAttribute('aria-labelledby', 'test-id title-id');
+});
```

Comment:
What would you think about testing this using `toHaveAccessibleName`rather than explicitly checking for the attribute value with `toHaveAttribute`? IMO that would align closer with how this behavior is actually experienced by users.